packages/cli-repl/src/cli-repl.spec.ts

```diff
@@ -485,6 +485,60 @@ describe('CliRepl', () => {
           expect(output).not.to.include('oh no');
         });
       });
+
+      context('with a global configuration file', () => {
+        it('loads a global config file as YAML if present', async() => {
+          const globalConfigFile = path.join(tmpdir.path, 'globalconfig.conf');
+          await fs.writeFile(globalConfigFile, 'mongosh:\n  redactHistory: remove-redact');
+
+          cliReplOptions.globalConfigPaths = [ globalConfigFile ];
+          cliRepl = new CliRepl(cliReplOptions);
+          await cliRepl.start('', {});
+
+          output = '';
+          input.write('config.get("redactHistory")\n');
+          await waitEval(cliRepl.bus);
+          expect(output).to.include('remove-redact');
+        });
+
+        it('loads a global config file as EJSON if present', async() => {
+          const globalConfigFile = path.join(tmpdir.path, 'globalconfig.conf');
+          await fs.writeFile(globalConfigFile, '{ "redactHistory": "remove-redact" }');
+
+          cliReplOptions.globalConfigPaths = [ globalConfigFile ];
+          cliRepl = new CliRepl(cliReplOptions);
+          await cliRepl.start('', {});
+
+          output = '';
+          input.write('config.get("redactHistory")\n');
+          await waitEval(cliRepl.bus);
+          expect(output).to.include('remove-redact');
+        });
+
+        it('warns if a global config file is present but could not be parsed', async() => {
+          const globalConfigFile = path.join(tmpdir.path, 'globalconfig.conf');
+          await fs.writeFile(globalConfigFile, 'a: b: c\n');
+
+          cliReplOptions.globalConfigPaths = [ globalConfigFile ];
+          cliRepl = new CliRepl(cliReplOptions);
+          await cliRepl.start('', {});
+
+          expect(output).to.include('Could not parse global configuration file at');
+          expect(output).to.include('a: b: c'); // echoes back the offending line
+        });
+
+        it('warns if a global config file is present but its values are invalid', async() => {
+          const globalConfigFile = path.join(tmpdir.path, 'globalconfig.conf');
+          await fs.writeFile(globalConfigFile, 'mongosh:\n  redactHistory: meow');
+
+          cliReplOptions.globalConfigPaths = [ globalConfigFile ];
+          cliRepl = new CliRepl(cliReplOptions);
+          await cliRepl.start('', {});
+
+          expect(output).to.include('Warning: Ignoring config option "redactHistory" from');
+          expect(output).to.include("redactHistory must be one of 'keep', 'remove', or 'remove-redact'");
+        });
+      });
```

Comment:
Do we have a test that ensures the options from the global conf are overiden by the user's conf?

Reply:
No, thank you for catching that! Added this in 8052b5f07bad4b1f4484643f19f902a7ef3544fc